app/src/main/java/ch/epfl/cs311/wanderwave/model/spotify/SpotifyController.kt

```diff
@@ -305,6 +307,87 @@ constructor(
     }
   }
 
+  /**
+   * Retrieve the image URL of an album.
+   *
+   * @param albumId The Spotify ID of the album.
+   * @return A Flow that emits the album image URL as a string.
+   */
+  fun getAlbumImage(albumId: String): Flow<String?> {
+    return callbackFlow {
+      val accessToken = authenticationController.getAccessToken()
+
+      if (accessToken == null) {
+        trySend(null)
+        close()
+        return@callbackFlow
+      }
+
+      val urlString = "https://api.spotify.com/v1/albums/$albumId"
+      val response = authenticationController.makeApiRequest(URL(urlString))
+
+      withContext(Dispatchers.IO) {
+        try {
+          if (response != "FAILURE") {
+            val jsonObject = JSONObject(response)
+            val images = jsonObject.getJSONArray("images")
+            if (images.length() > 0) {
+              val imageUrl = images.getJSONObject(0).getString("url")
+              trySend(imageUrl)
+            } else {
+              trySend(null)
+            }
+          } else {
+            trySend(null)
+          }
+        } catch (e: Exception) {
+          Log.e("SpotifyController", "Error fetching album image: ${e.message}")
+          trySend(null)
+        }
+      }
+
+      awaitClose {}
+    }
+  }
+
+  /**
+   * Download an image from a URL and display it in an ImageView.
+   *
+   * @param imageUrl The URL of the image.
+   * @param imageView The ImageView to display the image in.
+   */
+  fun downloadAndDisplayImage(imageUrl: String, imageView: ImageView) {
```

Comment:
it's a detail but maybe here pass the scope as an argument, as in the future we will use it in the viewmodel, we will pass the scope here, and it's good in my opinion to have the same scope everywhere

Reply:
done